unit_tests/fields/field_python_test.cpp

```diff
@@ -79,151 +89,179 @@ string input = R"INPUT(
-    Input::Record in_rec=reader.get_root_interface<Input::Record>();
-    UnitSI unit = UnitSI().m();
-    FieldAlgoBaseInitData init_data("field_python", 3, unit);
-
-    auto flux=VectorField::function_factory(in_rec.val<Input::AbstractRecord>("field_string"), init_data);
-    {
-        Space<3>::Point point_1, point_2;
-        point_1(0)=1.0; point_1(1)= pi / 2.0; point_1(2)=1.0;
-        point_2(0)= sqrt(2.0); point_2(1)= 3.0 * pi / 4.0; point_2(2)= pi / 2.0;
-
-        ElementAccessor<3> elm;
-        arma::vec3 result;
-
-        result = flux->value( point_1, elm);
-        EXPECT_DOUBLE_EQ( cos(pi /2.0 ) , result[0]); // should be 0.0
-        EXPECT_DOUBLE_EQ( 1, result[1]);
-        EXPECT_DOUBLE_EQ( 1, result[2]);
-
-        result = flux->value( point_2, elm);
-        EXPECT_DOUBLE_EQ( -1, result[0]);
-        EXPECT_DOUBLE_EQ( 1, result[1]);
-        EXPECT_DOUBLE_EQ( 1, result[2]);
-    }
-
-    auto flux_unit_conv=VectorField::function_factory(in_rec.val<Input::AbstractRecord>("field_string_unit_conversion"), init_data);
-    {
-        Space<3>::Point point_1, point_2;
-        point_1(0)=1.0; point_1(1)= pi / 2.0; point_1(2)=1.0;
-        point_2(0)= sqrt(2.0); point_2(1)= 3.0 * pi / 4.0; point_2(2)= pi / 2.0;
-
-        ElementAccessor<3> elm;
-        arma::vec3 result;
-
-        result = flux_unit_conv->value( point_1, elm);
-        EXPECT_DOUBLE_EQ( 0.01*cos(pi /2.0 ) , result[0]); // should be 0.0
-        EXPECT_DOUBLE_EQ( 0.01, result[1]);
-        EXPECT_DOUBLE_EQ( 0.01, result[2]);
-
-        result = flux_unit_conv->value( point_2, elm);
-        EXPECT_DOUBLE_EQ( -0.01, result[0]);
-        EXPECT_DOUBLE_EQ( 0.01, result[1]);
-        EXPECT_DOUBLE_EQ( 0.01, result[2]);
-    }
-
-    auto conc=ScalarField::function_factory(in_rec.val<Input::AbstractRecord>("field_file"), init_data);
-    {
-        Space<3>::Point point_1, point_2;
-        point_1(0)=1; point_1(1)=0; point_1(2)=0;
-        point_2(0)=1; point_2(1)=2; point_2(2)=3;
-        ElementAccessor<3> elm;
-
-        EXPECT_EQ( 0, conc->value(point_1, elm));
-        EXPECT_EQ( 6, conc->value(point_2, elm));
-    }
-
-
-}
-
-TEST(FieldPython, python_exception) {
-    FieldPython<3, FieldValue<3>::Scalar> scalar_func;
-	EXPECT_THROW_WHAT( { scalar_func.set_python_field_from_string(python_function, "func_xxx"); }, PythonLoader::ExcPythonError,
-        "Message: module 'python_field_func_xxx' has no attribute 'func_xxx'");
-
-}
-
-
-TEST(FieldPython, call_object_error) {
-    FieldPython<3, FieldValue<3>::Scalar> scalar_func;
-	EXPECT_THROW( { scalar_func.set_python_field_from_string(python_call_object_err, "func_xyz"); }, PythonLoader::ExcPythonError);
-        //"Program Error: Python Error: func_xyz() takes exactly 4 arguments (3 given)"
-
-}
-
-
-#else
-TEST(FieldPython, python_not_supported) {
-
+    FieldPython<3, FieldValue<3>::Vector> vec_func;
+    vec_func.set_python_field_from_class("fields/field_python_class.py", "PythonAsm");
```

Comment:
Test of FieldPython according to new assembly.

Reply:
Will be solve in #1488